### ISSUE TITLE
idea: Support for adding to the summary page of a workflow run

### ISSUE THREAD

**Issue by jabbermouth** [borchero/terraform-plan-comment#13]
Add ability to post the markdown as a job summary: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary

**Comment by RDhar** [comment]
@jabbermouth Something like [this](https://github.com/DevSecTop/TF-via-PR/actions/runs/9664099154#summary-26657936544)?

![image](https://github.com/borchero/terraform-plan-comment/assets/19497993/912b153b-2b37-422b-9171-e5a324229e20)
Screenshot of TF plan in a GitHub Action workflow's job summary.

**Comment by jabbermouth** [comment]
Yep 👍  Basically, what goes in the PR but on the run summary page.

**Comment by RDhar** [comment]
Might be worth giving [TF-via-PR](https://github.com/DevSecTop/TF-via-PR?tab=readme-ov-file#readme) GitHub Action a go; the above screenshot is taken directly from a recent workflow run to [demo a realistic use-case](https://github.com/DevSecTop/TF-via-PR/actions/runs/9664099154#summary-26657936544).

You can probably tell I'm the author, which is why I'm on the look-out for use-cases/feature-requests that users are seeking for these sort of Actions.

**Comment by jabbermouth** [comment]
Thanks, I'll take a look.

**Comment by borchero** [comment]
This sounds very reasonable! And I think it can be nicely combined with the request in #12 :eyes:

**Comment by joryirving** [comment]
I would love to see this. Using a single action to print to both comments on a PR, and summary on a run without a PR would be great.

**Comment by rdhar** [comment]
@joryirving You're in luck, both are [supported here](https://github.com/DevSecTop/TF-via-PR?tab=readme-ov-file#readme)!

**Comment by joryirving** [comment]
Sure, but that's a different action than this repo. I'd rather not refactor for a new action if I don't need to.

**Comment by rdhar** [comment]
@joryirving That's totally fair. Might be pleasantly surprised by how little there is to "refactor", but I understand replacing and trialling a GH Action takes time.

Worth noting, it does a li'l more than just render a plan as a comment—it streamlines Terraform/Tofu automation as a whole. Attached a screenshot below of a workflow (matrix) job jummary, as per this issue request.

![Workflow job summary with plan file artifact.](https://raw.githubusercontent.com/devsectop/tf-via-pr/refs/heads/main/.github/assets/workflow.png)

**Comment by joryirving** [comment]
It honestly feels weird, advertising your GHA in another actions repo. That feels inherently wrong, and actively makes me not want to switch.

**Comment by mircea-pavel-anton** [comment]
@rdhar I would also chime in here for a moment to give my 2 cents that what you're doing is not really constructive.

This is an issue opened in *this* repo to suggest/request a new feature for *this* project. Promoting your own project (even if it does answer the initial request) multiple times just pollutes this thread that had an entirely different purpose to begin with.

**Comment by rdhar** [comment]
Hey @joryirving and @mircea-pavel-anton, thanks for sharing your thoughts. I get why it feels off to mention my project here, but I just wanted to share an alternative for the feature request since it's been open for nearly four months now.

Open-source thrives on communities which share ideas and explore different options, instead of limiting ourselves to just one solution. I appreciate your perspective as well, and hope you see where I'm coming from.

**Comment by joryirving** [comment]
To be honest, the Summary would be a "nice to have" but not a requirement. I really like how this GHA formats the comments https://github.com/borchero/terraform-plan-comment#example-comments
